### PR TITLE
sensors: prevent double orb_copy of gyro topic

### PR DESCRIPTION
**Pull request by julianoes**
By using the uORB::Subscription API we use a separate subscription
rather than `orb_copy` on the existing file descriptor used in
sensors through `px4_poll`.

This fixes a very peculiar problem that we observed in SITL in CI for
fixedwing. The events were as follows:

1. `sensors` does `px4_poll` on the gyro topic (as normal), and gets
   the latest sample using `orb_copy`.
2. A parameter update happens when the mag is initialized and triggers
   `VotedSensorsUpdate::parameters_update()` where `orb_copy` happens
   before the main loop in `sensors` has started a `px4_poll`.
3. `sensors` now does the `px4_poll`, however waits indefinitely because
   it has already copied the latest sample.
   Also, the `px4_poll` will never time out because in lockstep the
   simulator waits for the next actuator control message which it never
   gets and therefore it never sends the next sensor message with a new
   timestamp to advance the time.

This only happens for fixedwing because there is only one "uORB path"
through the system unlike for multicopter where a gyro sample can get
picked up by either `sensors` or directly `mc_att_control`, so the
system can survive if `sensors` has "drops".

### CODE REVIEW

**Comment by dagar** [comment]
This also explains (at least partially) the estimator timeslip during param sync. "Stealing" an orb_copy means the sensors module misses an iteration, resulting in a gap in the data consumed by the estimator.

File: src/modules/sensors/voted_sensors_update.cpp [hunk 1]
```diff
@@ -178,21 +177,19 @@ void VotedSensorsUpdate::parameters_update()
 	for (int topic_instance = 0; topic_instance < ACCEL_COUNT_MAX; ++topic_instance) {
 
 		if (topic_instance < _accel.subscription_count) {
-			// valid subscription, so get the driver id by getting the published sensor data
-			sensor_accel_s report;
 
-			if (orb_copy(ORB_ID(sensor_accel), _accel.subscription[topic_instance], &report) == 0) {
-				int temp = _temperature_compensation.set_sensor_id_accel(report.device_id, topic_instance);
+			uORB::Subscription<sensor_accel_s> report{ORB_ID(sensor_accel), 0};
```
Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]:
```suggestion
			uORB::Subscription<sensor_accel_s> report{ORB_ID(sensor_accel), topic_instance};
```

File: src/modules/sensors/voted_sensors_update.cpp [hunk 2]
```diff
@@ -178,21 +177,19 @@ void VotedSensorsUpdate::parameters_update()
 	for (int topic_instance = 0; topic_instance < ACCEL_COUNT_MAX; ++topic_instance) {
 
 		if (topic_instance < _accel.subscription_count) {
```
Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]: This logic is a bit odd, why not just loop from to _accel.subscription_count?

File: src/modules/sensors/voted_sensors_update.cpp [hunk 3]
```diff
@@ -178,21 +177,19 @@ void VotedSensorsUpdate::parameters_update()
 	for (int topic_instance = 0; topic_instance < ACCEL_COUNT_MAX; ++topic_instance) {
```
Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]:
```suggestion
	for (int topic_instance = 0; topic_instance < _accel.subscription_count; ++topic_instance) {
```

File: src/modules/sensors/voted_sensors_update.cpp (hunk 1 above)
Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]:
```suggestion
			uORB::Subscription<sensor_accel_s> report{ORB_ID(sensor_accel), topic_instance};
```

Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]:
```suggestion
			uORB::Subscription<sensor_accel_s> report{ORB_ID(sensor_accel), 0, topic_instance};
```

File: src/modules/sensors/voted_sensors_update.cpp [hunk 4]
```diff
@@ -201,21 +198,19 @@ void VotedSensorsUpdate::parameters_update()
 	for (int topic_instance = 0; topic_instance < BARO_COUNT_MAX; ++topic_instance) {
 
 		if (topic_instance < _baro.subscription_count) {
-			// valid subscription, so get the driver id by getting the published sensor data
-			sensor_baro_s report;
 
-			if (orb_copy(ORB_ID(sensor_baro), _baro.subscription[topic_instance], &report) == 0) {
-				int temp = _temperature_compensation.set_sensor_id_baro(report.device_id, topic_instance);
+			uORB::Subscription<sensor_baro_s> report{ORB_ID(sensor_baro), 0};
```
Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]:
```suggestion
			uORB::Subscription<sensor_baro_s> report{ORB_ID(sensor_baro), 0, topic_instance};
```

File: src/modules/sensors/voted_sensors_update.cpp [hunk 5]
```diff
@@ -154,21 +155,19 @@ void VotedSensorsUpdate::parameters_update()
 	for (int topic_instance = 0; topic_instance < GYRO_COUNT_MAX; ++topic_instance) {
 
 		if (topic_instance < _gyro.subscription_count) {
-			// valid subscription, so get the driver id by getting the published sensor data
-			sensor_gyro_s report;
 
-			if (orb_copy(ORB_ID(sensor_gyro), _gyro.subscription[topic_instance], &report) == 0) {
-				int temp = _temperature_compensation.set_sensor_id_gyro(report.device_id, topic_instance);
+			uORB::Subscription<sensor_gyro_s> report{ORB_ID(sensor_gyro), 0};
```
Comment (dagar) [inline src/modules/sensors/voted_sensors_update.cpp]:
```suggestion
			uORB::Subscription<sensor_gyro_s> report{ORB_ID(sensor_gyro), 0, topic_instance};
```

**Comment by julianoes** [comment]
@dagar you can just add a commit with your changes, thx.

**Comment by lamping7** [comment]
Re-enable FW in this PR and grind it or after?

**Comment by julianoes** [comment]
> Re-enable FW in this PR and grind it or after?

Right, we shouldn't forget!

**Comment by dagar** [comment]
So far so good, rebasing to force the build again.

**Comment by dagar** [comment]
Great debugging @julianoes

**Pull request merged**